libbeat/common/field.go

```diff
@@ -292,6 +292,46 @@ func (f Fields) getKeys(namespace string) []string {
 	return keys
 }
 
+// canConcat checks if inside fields the given node can be concatenated.
+// If the node already exists, this is only allowed for nodes of type object.
+func (f Fields) CanConcat(key string) bool {
+	keys := strings.Split(key, ".")
+	return f.canConcat(keys)
+}
+
+func (f Fields) canConcat(keys []string) bool {
+
+	// Nothing to compare, so does not contain it
+	if len(keys) == 0 {
+		return false
+	}
+
+	key := keys[0]
+	if key == "" {
+		return false
+	}
+	keys = keys[1:]
+
+	for _, field := range f {
+
+		if field.Name == key {
+
+			//// It's the last key to compare
+			if len(keys) == 0 {
+				return false
+			}
```

Comment:
nit: 
- `////`
- lets not add extra whitespace between for and if statements or beginning of function body. In most cases indentation should be enough to distinguish blocks.